app/features/bookmarks/BookmarkForm.tsx

```diff
@@ -11,13 +11,16 @@ interface BookmarkFormProps {
 }
 const IMAGE_PLACEHOLDER = "https://via.placeholder.com/450?text=Enter+an+image+url";
 export function BookmarkForm({ initial }: BookmarkFormProps) {
-  let form = useValidatedForm(initial);
-  let navigate = useNavigate();
-  let [image, setImage] = useState(initial?.image || IMAGE_PLACEHOLDER);
+  const form = useValidatedForm(initial);
+  const transition = useTransition();
+  const navigate = useNavigate();
+  const [image, setImage] = useState(initial?.image || IMAGE_PLACEHOLDER);
+  const isProcessing = transition.state === "submitting";
```

Comment:
Scoped this just to `=== "submitting"` instead of `!== "idle"` because the latter will trip, in this case, when "Cancel" is clicked.  E.g. `"loading"`.